app/src/main/java/com/lucas/medicaltools/repository/MedicalToolsRepository.kt

```diff
@@ -0,0 +1,29 @@
+package com.lucas.medicaltools.repository
+
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medicaltools.cache.DataSource
+import com.lucas.medicaltools.cache.DiskDataSource
+import com.lucas.medicaltools.cache.MemoryDataSource
+import com.lucas.medicaltools.cache.NetworkDataSource
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Response
+
+class MedicalToolsRepository() {
+    val dataSource: DataSource = DataSource(MemoryDataSource(), DiskDataSource(), NetworkDataSource())
```

Comment:
cache = Response<MutableList<MedicalTool>>>?

if (cache != null) return observable.just(cache)
else return networkApi.getMedication().doOnNext { cache = it}

Reply:
don't forget the lateinit